doc/crawler.rst

```diff
@@ -0,0 +1,91 @@
+Crawler -- mm2_crawler
+======================
+
+Overview
+--------
+
+The crawler scans all available mirrors if their content is up to date. The
+crawler only scans mirrors which are not disabled (host.user_active,
+host.admin_active, site.user_active, site.admin_active).
+
+The crawler retrieves the list of active mirrors from the database and
+spawns a thread for each host. The crawling of a host consists of a loop
+over all categories which have been configured for this host and one category
+is crawled after another.
+
+The crawler has the possibility to crawl a category via HTTP, FTP or RSYNC.
+For each category the protocol to crawl is selected from the available
+category protocols. RSYNC has the highest priority and is followed by HTTP
+and the protocol used last (if no other protocol is available) is FTP.
```

Comment:
Good question about the protocol order. This is probably related that most mirrors support HTTP and that FTP is not the best protocol when it comes to firewalls. Looking at the code of per_host() the crawler first tries to scan a whole category using RSYNC and if that fails does a whole directory using FTP and the last option is each single file via HTTP.
But then there is the function which selects the preferred protocol which has the ordering RSYNC -> HTTP -> FTP. I think this function could be from me.  The reason I put FTP at last probably comes from dislike of FTP as a mirror admin. I would rather get rid of it. But maybe this needs to be re-evaluated.